### ISSUE TITLE
Segfault during (Re)importing assets of converted project

### ISSUE THREAD

**Issue by jonri** [godotengine/godot#65014]
### Godot version

4.0 alpha 14; custom build 58f8f3a; 4.0 Beta 1

### System information

Arch Linux

### Issue description

I tried to convert my Godot 3 project to Godot 4 and edit it.  After conversion, it got to "(Re)importing assets" and crashed with a segfault during this process.  Opening the same project from the list afterwards yields the same result.

I'm not expecting the project to convert cleanly, but simpler projects have imported without crashing and I could fix the differences in nodes and scripts afterwards.

Console errors before the crash, and the backtrace:
[backtrace.txt](https://github.com/godotengine/godot/files/9441215/backtrace.txt)

The script errors aren't surprising, but the `!importer.is_valid()` sounded like it might be relevant.

### Steps to reproduce

1. git clone https://github.com/RTForge/DawnOfTheDuck
2. Using Godot 4, import the project in the `duck` subdirectory from the editor, or edit the project from the command line

### Minimal reproduction project

See above

**Comment by lyuma** [comment]
This is difficult to reproduce. It happens only 5%-10% of the time for me. It indeed still happens in RC2, even though many importer related issues were already solved.
TL;DR (see below): This seems to be a bug with the dynamic font importer.

Reproduction steps:
clone the attached project (yes, it's made for Godot 3.x. that's fine.)
Edit project.godot and remove `run/main_scene` from it. That makes it easier to exit out of the editor without having errors pop up.

Note: I was unable to reproduce the issue after porting the project to Godot 4.x. Here is my diff if you're interested: 
[duck_to_4.diff.txt](https://github.com/godotengine/godot/files/10741944/duck_to_4.diff.txt)


Then keep running this repeatedly, closing the editor each time it loads successfully.
```
rm -rf .godot; ~/Downloads/Godot_v4.0-rc2_win64.exe -w -e
```
when it crashes, it's usually very early on.
```
$ Vulkan API 1.3.224 - Forward+ - Using Vulkan Device #0: NVIDIA - NVIDIA GeForce RTX 3090

ERROR: Cannot open file 'res://tp/new_theme.tres'.
   at: (scene/resources/resource_format_text.cpp:1649)
ERROR: Failed loading resource: res://tp/new_theme.tres. Make sure resources have been imported by opening the project in the editor at least once.
   at: (core/io/resource_loader.cpp:221)
ERROR: Cannot open file 'res://.godot/imported/Raleway-Medium.ttf-bf3a59b6b84ee7d9b4ef514fb82fdb8e.fontdata'.
   at: (core/io/resource_format_binary.cpp:1249)
ERROR: Cannot open file 'res://.godot/imported/SigmarOne-Regular.ttf-c54f2e16d2b46652f2ad680fe092e574.fontdata'.
   at: (core/io/resource_format_binary.cpp:1249)
ERROR: Cannot open file 'res://.godot/imported/boat2.glb-fe26d62f6ca55c071a4b687d6a0f4ecc.scn'.
   at: (core/io/resource_format_binary.cpp:1249)
ERROR: Cannot open file 'res://.godot/imported/duck.glb-af6623336133712d01a6bfdefd9d6344.scn'.
   at: (core/io/resource_format_binary.cpp:1249)
ERROR: Cannot open file 'res://.godot/imported/gun.glb-05f42a236277268338275ecffb48440f.scn'.
   at: (core/io/resource_format_binary.cpp:1249)
ERROR: Cannot open file 'res://.godot/imported/muzzle_flash.obj-8f77222eab2a9102e995fbb8d9e55f50.mesh'.
   at: (core/io/resource_format_binary.cpp:1249)
ERROR: Cannot open file 'res://.godot/imported/tub.glb-b050456f185c0941f22da0a05b67312d.scn'.
   at: (core/io/resource_format_binary.cpp:1249)
ERROR: Cannot open file 'res://.godot/imported/duck_intro.ogg-75ff4c73f791e148447ced02042586f3.oggvorbisstr'.
   at: (core/io/resource_format_binary.cpp:1249)
ERROR: Cannot open file 'res://.godot/imported/engine.wav-5941c30789ec1b20ddeb5177648accc9.sample'.
   at: (core/io/resource_format_binary.cpp:1249)
ERROR: Cannot open file 'res://.godot/imported/faucet.wav-c64c418ac8b13b1dfbca9ee15a387a1f.sample'.
   at: (core/io/resource_format_binary.cpp:1249)
ERROR: Cannot open file 'res://.godot/imported/fireball.wav-d12de690fb1bc5635b3139e19432a5d6.sample'.
   at: (core/io/resource_format_binary.cpp:1249)
ERROR: Cannot open file 'res://.godot/imported/good_bad_ugly.wav-a1967205b565992397d00404d9fbd6b1.sample'.
   at: (core/io/resource_format_binary.cpp:1249)
ERROR: Cannot open file 'res://.godot/imported/gun1.wav-53ac4776464b0aabe9fd2135e1575e8f.sample'.
   at: (core/io/resource_format_binary.cpp:1249)
ERROR: Cannot open file 'res://.godot/imported/gun2.wav-a8181bf7dbe45fabe700db54c3f71de3.sample'.
   at: (core/io/resource_format_binary.cpp:1249)
ERROR: Cannot open file 'res://.godot/imported/gun3.wav-c8cddd93351d54c5fdfd782185777204.sample'.
   at: (core/io/resource_format_binary.cpp:1249)
ERROR: Cannot open file 'res://.godot/imported/gun4.wav-4df729fde5f1488da84543ae9e59c5f8.sample'.
   at: (core/io/resource_format_binary.cpp:1249)
ERROR: Cannot open file 'res://.godot/imported/metal1.wav-ee30351527ac3efe7f953556aeb67979.sample'.
   at: (core/io/resource_format_binary.cpp:1249)
ERROR: Cannot open file 'res://.godot/imported/metal2.wav-676d64d49af5dd9f0799238bfc25941c.sample'.
   at: (core/io/resource_format_binary.cpp:1249)
ERROR: Cannot open file 'res://.godot/imported/metal3.wav-0b8640b9535c0fa11254c2d6b8b7912d.sample'.
   at: (core/io/resource_format_binary.cpp:1249)
ERROR: Cannot open file 'res://.godot/imported/metal4.wav-2792b5a566923c1990b717071c307bc6.sample'.
   at: (core/io/resource_format_binary.cpp:1249)
ERROR: Cannot open file 'res://.godot/imported/splash.wav-ff0f95428d6fe31dc6a062bb450bc559.sample'.
   at: (core/io/resource_format_binary.cpp:1249)
Unicode parsing error, some characters were rep
[1]+  Segmentation fault      ~/Downloads/Godot_v4.0-rc1_win64.exe -w -e
```

I was able to reproduce the crash in both rc1 and also rc2.
The Unicode parsing error is not always displayed, but it does give a clue that perhaps it is related to fonts.

I was able to reproduce a crash from Visual Studio.
Here are the relevant all-thread stacktraces from two different crashing runs in a custom build on `4848877`, close to RC2. I used `Debug.ListCallStack /ShowLineOffset /AllThreads` to display stacktraces from all threads in Visual Studio's Command Window.
Crashing Run 1: [all_threads_stacktraces_fontdata_import.txt](https://github.com/godotengine/godot/files/10741868/all_threads_stacktraces_fontdata_import.txt)
Crashing Run 2: [all_threads_stacktraces_fontdata_import2.txt](https://github.com/godotengine/godot/files/10741896/all_threads_stacktraces_fontdata_import2.txt)

In the first one, two threads were importing fonts at the same time (`ResourceImporterDynamicFont::import`)

In the second stacktrace, only one was importing at that moment, but the other import thread was actively cleaning up from a .ttf import - this time I was able to print out the filenames in the debugger:
Thread 42: res://fonts/Raleway-Medium.ttf.import
Thread 43: res://.godot/imported/SigmarOne-Regular.ttf-c54f2e16d2b46652f2ad680fe092e574

Pinging @bruvzg since you might know something about font import

**Comment by lyuma** [comment]
For now, as a workaround, disable multithreaded import in project settings, or retry multiple times until all of the problematic files get imported and Godot moves on. I think at least one file gets successfully imported each time this happens.

**Comment by jonri** [comment]
I gave my original reproduction steps a try with the official 4.0.3 and 4.1-beta 3 release packages.

4.0.3:  No crash during import, dependency error (without crash) when editing the project afterwards.  Closing and re-opening the project opens the converted project in the expected form.
4.1-beta3: No crash during import, and editing the project opens it in the expected form without the error from 4.0.3

It seems to me like the original segfault is no longer happening, and the intermediate dependency error in 4.0.3 isn't worth worrying about.
